repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java

```diff
@@ -38,41 +37,34 @@
     String httpMethod;
     String requestURI;
     String requestBody;
-    HashMap<String, String> headers;
-
-    public RequestLog(MutableHttpServletRequest mutableHttpServletRequest,
-                      FilterContext filterContext) throws IOException {
-
-        Filter filter = filterContext.getFilterConfig();
+    Map<String, String> headers;
 
+    public RequestLog(HttpServletRequest httpServletRequest, Filter filter) throws IOException {
         preamble = "Intrafilter Request Log";
         timestamp = new DateTime().toString();
         currentFilter = StringUtils.isEmpty(filter.getId()) ? filter.getName() : filter.getId() + "-" + filter.getName();
-        httpMethod = mutableHttpServletRequest.getMethod();
-        requestURI = mutableHttpServletRequest.getRequestURI();
-        headers = convertRequestHeadersToMap(mutableHttpServletRequest);
+        httpMethod = httpServletRequest.getMethod();
+        requestURI = httpServletRequest.getRequestURI();
+        headers = convertRequestHeadersToMap(httpServletRequest);
 
-        //Have to wrap the input stream in something that can be buffered, as well as reset.
-        ServletInputStream bin = mutableHttpServletRequest.getInputStream();
-        bin.mark(Integer.MAX_VALUE); //Something doesn't support mark reset
-        requestBody = IOUtils.toString(bin); //http://stackoverflow.com/a/309448
-        bin.reset();
+        ServletInputStream inputStream = httpServletRequest.getInputStream();
+        if (inputStream.markSupported()) {
+            inputStream.mark(Integer.MAX_VALUE);
+            requestBody = IOUtils.toString(inputStream); //http://stackoverflow.com/a/309448
+            inputStream.reset();
+        } else {
+            throw new RuntimeException("Servlet input stream does not support mark/reset: " + inputStream);
```

Comment:
Seems legit to throw an exception to me, but not directly `RuntimeException` some inheritor.